TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/activities/ProfileActivity.java

```diff
@@ -0,0 +1,119 @@
+package com.fbu.thefoodienetwork.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.fbu.thefoodienetwork.R;
+import com.fbu.thefoodienetwork.adapters.ReviewAdapter;
+import com.fbu.thefoodienetwork.databinding.ActivityProfileBinding;
+import com.fbu.thefoodienetwork.keys.UserKeys;
+import com.fbu.thefoodienetwork.models.ParseReview;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+//TODO change LoginActivity goMainActivity when finish
+public class ProfileActivity extends AppCompatActivity {
+    private final static String TAG = "ProfileActivity";
+    private ActivityProfileBinding binding;
+    private ParseUser user = ParseUser.getCurrentUser();
+    private ReviewAdapter reviewAdapter;
+    private List<ParseReview> allReviews;
+    private LinearLayoutManager layoutManager;
+    private RecyclerView reviewRecyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityProfileBinding.inflate(getLayoutInflater());
+        View view = binding.getRoot();
+        setContentView(view);
+
+        reviewRecyclerView = binding.postsRecyclerView;
+        allReviews = new ArrayList<>();
+        reviewAdapter = new ReviewAdapter(this, allReviews);
+        reviewRecyclerView.setAdapter(reviewAdapter);
+        layoutManager = new LinearLayoutManager(this);
+        reviewRecyclerView.setLayoutManager(layoutManager);
+
+        queryReviews();
+
+        binding.editButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                goEditProfileActivity();
+            }
+        });
+
+        try {
+
+            if (user.getString(UserKeys.screenName) == null || user.getString(UserKeys.screenName).trim().equals("")) {
+                binding.textViewScreenName.setText(user.getUsername());
+            } else {
+                binding.textViewScreenName.setText(user.getString(UserKeys.screenName));
+                binding.textViewUsername.setText(user.getUsername());
+            }
+
+
+            try {
+                binding.bioTextView.setText(user.getString(UserKeys.bio));
```

Comment:
Great usage of keys!  In this case, it's convention to use all caps for string constants, though.